src/tactic/ui/widget/discussion_wdg.py

```diff
@@ -1538,35 +1473,160 @@ def get_context_wdg(my, process, context, show_context_notes=True):
+
+        return div
+
+
+
+__all__.append("NotesContextWdg")
+
+class NotesContextWdg(BaseRefreshWdg):
+
+    def get_display(my):
+        notes = my.kwargs.get("notes")
+        note_keys = my.kwargs.get("note_keys")
+        if note_keys:
+            notes = Search.get_by_search_keys(note_keys)
+
+        my.default_num_notes = my.kwargs.get("default_num_notes")
+        my.note_expandable = my.kwargs.get("note_expandable")
+        my.show_note_status = my.kwargs.get("show_note_status")
+
+        my.attachments = my.kwargs.get("attachments")
+
+
+        if my.attachments == None:
+            from pyasm.biz import Snapshot
+            snapshots = Snapshot.get_by_sobjects(notes)
+            my.attachments = {}
+            for snapshot in snapshots:
+                parent_key = snapshot.get_parent_search_key()
+                xx = my.attachments.get(parent_key)
+                if not xx:
+                    xx = []
+                    my.attachments[parent_key] = xx
+                xx.append(snapshot)
+
+
+        elif not my.attachments or my.attachments == "{}":
+            my.attachments = {}
+
+
+
+
+        if my.show_note_status:
+            my.note_status_dict = ProdSetting.get_dict_by_key('note_status')
+        else:
+            my.note_status_dict = {}
+
+        my.note_format = my.kwargs.get("note_format")
+
+
+        div = DivWdg()
+
+        context_count = 0
+
+        for i, note in enumerate(notes):
 
+            note_content = DivWdg()
+            div.add(note_content)
+            note_content.add_style("max-height: 500px")
+            note_content.add_style("overflow-y: auto")
+            note_content.add_style("overflow-x: hidden")
+
+            if my.default_num_notes == -1:
+                note_hidden = True
+            elif context_count >= my.default_num_notes:
+                note_hidden = True
+            else:
+                note_hidden = False
+
+            note_wdg = my.get_note_wdg(note, note_hidden=note_hidden)
+            if i % 2 == 0:
+                note_wdg.add_color("background", "background", -3)
+            else:
+                note_wdg.add_color("background", "background", -6)
+            note_wdg.add_style("border-style: solid")
+            note_wdg.add_style("border-width: 0 0 1px 0")
+            note_wdg.add_style("border-color: %s" % div.get_color("table_border"))
+
+            note_content.add(note_wdg)
+            note_wdg.add_style("width: 395px")
+
+            context_count += 1
+
+
+        return div
+
+
+
+    def get_note_wdg(my, note, note_hidden=False):
+        div = DivWdg()
+        widget = NoteWdg(
+            note=note,
+
+            note_hidden=note_hidden,
+
+            note_expandable=my.note_expandable,
+            show_note_status=my.show_note_status,
+            note_format=my.note_format,
+            attachments=my.attachments,
+
+        )
+
+        div.add(widget)
         return div
 
 
 
 
+class NoteWdg(BaseRefreshWdg):
+
```

Comment:
add a comment to it to read something like
'''Display of a single note used by NoteCollectionWdg'''